examples/generate_random_structures/get_heteropolymer_structure.py

```diff
@@ -0,0 +1,140 @@
+c_sigmas = {"bb_sigma": 1.2 * sigma, "sc_sigma": 1.2 * sigma}
+
+epsilon = 0.1 * unit.kilojoule_per_mole
+a_epsilons = {"bb_eps": epsilon, "sc_eps": epsilon}
+b_epsilons = {"bb_eps": 2 * epsilon, "sc_eps": 2 * epsilon}
+c_epsilons = {"bb_eps": 3 * epsilon, "sc_eps": 3 * epsilon}
+
+# Bond angle definitions
+bond_angle_force_constant = 15.0 * unit.kilojoule_per_mole / unit.radian / unit.radian
+bond_angle_force_constants = {
+    "bb_bb_bb_angle_k": bond_angle_force_constant,
+    "bb_bb_sc_angle_k": bond_angle_force_constant,
+    "bb_sc_sc_angle_k": bond_angle_force_constant,
+}
+
+bb_bb_bb_equil_bond_angle = 120.0 * unit.degrees
+bb_bb_sc_equil_bond_angle = 120.0 * unit.degrees
+bb_sc_sc_equil_bond_angle = 120.0 * unit.degrees
+equil_bond_angles = {
+    "bb_bb_bb_angle_0": bb_bb_bb_equil_bond_angle,
+    "bb_bb_sc_angle_0": bb_bb_sc_equil_bond_angle,
+    "bb_sc_sc_angle_0": bb_bb_sc_equil_bond_angle,
+}
+
+# Torsion angle definitions
+torsion_force_constant = 0 * unit.kilojoule_per_mole
+torsion_force_constants = {
+    "bb_bb_bb_bb_torsion_k": torsion_force_constant,
+    "bb_bb_bb_sc_torsion_k": torsion_force_constant,
+    "bb_bb_sc_sc_torsion_k": torsion_force_constant,
+}
+
+
+equil_torsion_angle = 0.0 * unit.degrees
+equil_torsion_angles = {
+    "bb_bb_bb_bb_torsion_0": equil_torsion_angle,
+    "bb_bb_bb_sc_torsion_0": equil_torsion_angle,
+    "bb_bb_sc_sc_torsion_0": equil_torsion_angle,
+}
+torsion_periodicities = {"bb_bb_bb_bb_period": 3, "bb_bb_bb_sc_period": 1, "bb_bb_sc_sc_period": 1}
+
+A = {
+    "monomer_name": "A",
+    "backbone_length": 1,
+    "sidechain_length": 1,
+    "sidechain_positions": [0],
+    "bond_lengths": a_bond_lengths,
+    "epsilons": a_epsilons,
+    "sigmas": a_sigmas,
+}
+B = {
+    "monomer_name": "B",
+    "backbone_length": 2,
+    "sidechain_length": 1,
+    "sidechain_positions": [0],
+    "bond_lengths": b_bond_lengths,
+    "epsilons": b_epsilons,
+    "sigmas": b_sigmas,
+}
+C = {
+    "monomer_name": "C",
+    "backbone_length": 3,
+    "sidechain_length": 1,
+    "sidechain_positions": 1,
+    "bond_lengths": c_bond_lengths,
+    "epsilons": c_epsilons,
+    "sigmas": c_sigmas,
+}
+
+sequence = 4 * [A, B, C]
+
+# Build a coarse grained model
+cgmodel = CGModel(
+    masses=masses,
+    bond_lengths=a_bond_lengths,
+    bond_force_constants=bond_force_constants,
+    bond_angle_force_constants=bond_angle_force_constants,
+    torsion_force_constants=torsion_force_constants,
+    equil_bond_angles=equil_bond_angles,
+    equil_torsion_angles=equil_torsion_angles,
+    torsion_periodicities=torsion_periodicities,
+    include_nonbonded_forces=include_nonbonded_forces,
+    include_bond_forces=include_bond_forces,
+    include_bond_angle_forces=include_bond_angle_forces,
+    include_torsion_forces=include_torsion_forces,
+    constrain_bonds=constrain_bonds,
+    random_positions=random_positions,
+    sequence=sequence,
+    monomer_types=[A, B, C],
```

Comment:
Could we have sequence just be a list of strings ['A','B','C','A',...] instead of list of dictionaries so we don't need to pass all of the info for every monomer? It seems that all the detailed info we need is contained in monomer_types.

Reply:
Good question.  It could be done, but internally, then one would have to translate with a dictionary linking monomer types to monomer characters.  As it is, this translation doesn't need to be done.  It's not really that big of overhead, since I'm pretty sure python is passing by reference anyway (i.e. [A,B,C] I think is a list of pointers to the dictionary members. And it's not that large an amount of data.  But I'm definitely open to switching if it seems like there are other advantages.

Reply:
Ah ok - in that case I think it's fine to leave sequence as is for now.